app/src/main/java/com/henagle/www/kanban/MainActivity.java

```diff
@@ -1,133 +1,87 @@
+import android.support.v4.view.ViewPager;
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
-import android.view.LayoutInflater;
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
-import android.view.ViewGroup;
-import android.widget.BaseAdapter;
 import android.widget.EditText;
 import android.widget.LinearLayout;
-import android.widget.ListView;
-import android.widget.RelativeLayout;
-import android.widget.TextView;
 
 import com.henagle.www.kanban.model.Card;
-import com.henagle.www.kanban.model.CardSerializer;
+import com.henagle.www.kanban.model.DeckSerializer;
+import com.henagle.www.kanban.model.Deck;
+import com.henagle.www.kanban.model.Page;
 
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 public class MainActivity extends AppCompatActivity {
 
-    private ListView list;
-    private Map<Tab,List<Card>> tabLists;
-    private Map<Tab, CardListAdapter> adapters;
-    private Tab currentTab = Tab.TODO;
+    // Represents a map of Page -> Deck entries
+    // Source of truth for all card data
+    protected static Map<Page, Deck> decks;
 
+    private ViewPager viewPager;
+    private DeckPagerAdapter viewPagerAdapter;
     private AlertDialog addCardDialog;
 
-    public enum Tab {
-        TODO,
-        DOING,
-        DONE
-    }
-
-    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
-            = new BottomNavigationView.OnNavigationItemSelectedListener() {
-
-        @Override
-        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-            switch (item.getItemId()) {
-                case R.id.navigation_todo:
-                    list.setAdapter(adapters.get(Tab.TODO));
-                    currentTab = Tab.TODO;
-                    return true;
-                case R.id.navigation_doing:
-                    list.setAdapter(adapters.get(Tab.DOING));
-                    currentTab = Tab.DOING;
-                    return true;
-                case R.id.navigation_done:
-                    list.setAdapter(adapters.get(Tab.DONE));
-                    currentTab = Tab.DONE;
-                    return true;
-            }
-            return false;
-        }
-    };
-
     /**
-     * This is to handle the delete button being clicked on an item
-     * @param v
+     * Handle card deletion by delegating to the current fragment
+     * @param view view of Card to be deleted
      */
-    public void deleteClickHandler(View v) {
-        RelativeLayout vwParentRow = (RelativeLayout)v.getParent();
-        int index = ((ViewGroup)vwParentRow.getParent()).indexOfChild(vwParentRow);
-        List<Card> currentList = tabLists.get(currentTab);
-        if (currentList.size() > index) {
-            currentList.remove(index);
-        }
-        adapters.get(currentTab).notifyDataSetChanged();
```

Comment:
I moved this code into the `deleteClickHandler` method of the `DeckFragment` class.